controllers/internal/utils/dns_test.go

```diff
@@ -46,21 +54,51 @@ func TestEmptyFQDNButValidEdgeDNS(t *testing.T) {
 	if !connected() {
 		t.Skipf("no connectivity, skipping")
 	}
-	edgeDNSServer := "8.8.8.8"
 	fqdn := ""
 	// act
-	result, err := Dig(edgeDNSServer, fqdn)
+	result, err := Dig(fqdn, defaultEdgeDNSServer)
 	// assert
 	assert.NoError(t, err)
 	assert.Nil(t, result)
 }
 
 func TestEmptyEdgeDNS(t *testing.T) {
 	// arrange
-	edgeDNSServer := ""
 	fqdn := "whatever"
 	// act
-	result, err := Dig(edgeDNSServer, fqdn)
+	result, err := Dig(fqdn, DNSServer{Host: "", Port: 53})
+	// assert
+	assert.Error(t, err)
+	assert.Nil(t, result)
+}
+
```

Comment:
Happy new tests were added, please cover corner cases for: 

- TestEmptyEdgeDNSInTheList - empty list testing guardian condition 
- TestMultipleEdgeDNSInTheList - we are testing only one and zero valid edgeDNS servers, this test wouldcover more